pkg/library/flatten/merge.go

```diff
@@ -90,8 +90,10 @@ func mergeVolumeComponents(main, parent *dw.DevWorkspaceTemplateSpecContent, plu
 
 func mergeVolume(into, from *dw.VolumeComponent) error {
 	// If the new volume is persistent, make the original persistent
-	if !from.Ephemeral {
-		into.Ephemeral = false
+	if from.Ephemeral == nil {
+		into.Ephemeral = nil
+	} else {
+		from = into
```

Comment:
This might unset unset the ephemeral volume value (we don't want a later ephemeral volume making the persistent volume ephemeral).